cpp/tests/test_ide_secir_ageres.cpp

```diff
@@ -0,0 +1,159 @@
+        .set_distribution_parameter(4.0);
+
+    model.parameters.get<mio::isecir::TransitionDistributions>()[mio::AgeGroup(2)] = vec_delaydistrib3;
+
+    //TransitionProbabilities.
+    for (auto group = mio::AgeGroup(0); group < mio::AgeGroup(num_agegroups); ++group) {
+        std::vector<ScalarType> vec_prob(num_transitions, 0.5);
+        // The following probabilities must be 1, as there is no other way to go.
+        vec_prob[Eigen::Index(mio::isecir::InfectionTransition::SusceptibleToExposed)]        = 1;
+        vec_prob[Eigen::Index(mio::isecir::InfectionTransition::ExposedToInfectedNoSymptoms)] = 1;
+        model.parameters.get<mio::isecir::TransitionProbabilities>()[group]                   = vec_prob;
+    }
+
+    // Contact matrix.
+    mio::ContactMatrixGroup contact_matrix = mio::ContactMatrixGroup(1, num_agegroups);
+    contact_matrix[0] = mio::ContactMatrix(Eigen::MatrixXd::Constant(num_agegroups, num_agegroups, 10.));
+    model.parameters.get<mio::isecir::ContactPatterns>() = mio::UncertainContactMatrix(contact_matrix);
+
+    mio::ExponentialSurvivalFunction exponential(0.5);
+    mio::StateAgeFunctionWrapper prob(exponential);
+    for (auto group = mio::AgeGroup(0); group < mio::AgeGroup(num_agegroups); ++group) {
+        model.parameters.get<mio::isecir::TransmissionProbabilityOnContact>()[group] = prob;
+        model.parameters.get<mio::isecir::RelativeTransmissionNoSymptoms>()[group]   = prob;
+        model.parameters.get<mio::isecir::RiskOfInfectionFromSymptomatic>()[group]   = prob;
+    }
+    model.parameters.set<mio::isecir::Seasonality>(0.1);
+    // Start the simulation on the 40th day of a year (i.e. in February).
+    model.parameters.set<mio::isecir::StartDay>(40);
+
+    model.check_constraints(dt);
+
+    // Compare compartments at last time point with results from a previous run that are given here.
+    mio::TimeSeries<ScalarType> compartments = simulate(tmax, dt, model);
+    mio::Vector<ScalarType> compare_compartments(num_compartments * num_agegroups);
+    compare_compartments << 484.3056557672, 15.7685031055, 22.7020934123, 7.0615933479, 3.3491460693, 1.5803397070,
+        4454.5548070034, 10.6778615873, 484.3056557672, 31.0934010790, 21.1271954388, 23.6370809253, 3.9106794140,
+        1.7242153411, 4424.0110181177, 10.1907539167, 605.3820697090, 60.1973290710, 23.8046231705, 16.6085494134,
+        3.6307172673, 1.6536810707, 4278.2949856871, 10.4280446109;
+
+    int last_time_point = int(compartments.get_last_time());
+    ASSERT_EQ(compare_compartments.size(), static_cast<size_t>(compartments[last_time_point].size()));
+
+    for (int j = 1; j < compare_compartments.size(); j++) {
+        ASSERT_NEAR(compartments.get_last_value()[j], compare_compartments[j], 1e-7);
+    }
+
+    // Compare transitions at last time point with results from a previous run that are given here.
+
+    mio::isecir::Simulation sim(model, dt);
+    sim.advance(tmax);
```

Comment:
You are simulating twice in one test. I suggest to do one simulation as here above and then access the results for both compartments and transitions.